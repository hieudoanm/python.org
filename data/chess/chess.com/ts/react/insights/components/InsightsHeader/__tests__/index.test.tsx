import { mockResizeObserver } from '@web/utils/mock-resize-observer';
import { render } from '@testing-library/react';
import { InsightsHeader } from '..';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    asPath: '',
    basePath: '',
    pathname: '',
  }),
}));

jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockReturnValue({
    asPath: '',
    events: { on: jest.fn(), off: jest.fn() },
  }),
  usePathname: jest.fn().mockReturnValue(''),
  useSearchParams: jest.fn().mockReturnValue(new URLSearchParams()),
}));

describe('InsightsHeader', () => {
  beforeEach(() => {
    mockResizeObserver();
  });

  it('to match snapshot', () => {
    const { container } = render(
      <InsightsHeader name='' title='GM' avatar='' username='' />
    );
    expect(container).toMatchSnapshot();
  });
});
