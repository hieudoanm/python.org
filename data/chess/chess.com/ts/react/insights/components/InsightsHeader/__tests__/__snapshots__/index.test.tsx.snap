// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsHeader to match snapshot 1`] = `
<div>
  <header
    class="flex items-center justify-between gap-x-2"
  >
    <div
      class="flex items-center gap-x-2 md:gap-x-4"
    >
      <div
        class="aspect-square w-16 rounded-xl border bg-contain bg-center"
        style="background-image: url();"
      />
      <div>
        <div
          class="flex items-center gap-x-2"
        >
          <span
            class="badge badge-primary badge-outline"
          >
            GM
          </span>
          <div
            class="text-lg uppercase md:text-xl"
          />
          <div>
            <a
              href="https://www.chess.com/member/"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M215.5 224c29.2-18.4 48.5-50.9 48.5-88c0-57.4-46.6-104-104-104S56 78.6 56 136c0 37.1 19.4 69.6 48.5 88L96 224c-17.7 0-32 14.3-32 32c0 16.5 12.5 30 28.5 31.8L80 400l160 0L227.5 287.8c16-1.8 28.5-15.3 28.5-31.8c0-17.7-14.3-32-32-32l-8.5 0zM22.6 473.4c-4.2 4.2-6.6 10-6.6 16C16 501.9 26.1 512 38.6 512l242.7 0c12.5 0 22.6-10.1 22.6-22.6c0-6-2.4-11.8-6.6-16L256 432 64 432 22.6 473.4z"
                />
              </svg>
            </a>
          </div>
        </div>
        <p
          class="truncate"
        />
      </div>
    </div>
    <div
      class="join shadow"
    >
      <select
        class="join-item select select-bordered w-full"
        id="timeClass"
        name="timeClass"
      >
        <option
          value="daily"
        >
          Daily
        </option>
        <option
          value="rapid"
        >
          Rapid
        </option>
        <option
          value="blitz"
        >
          Blitz
        </option>
        <option
          value="bullet"
        >
          Bullet
        </option>
      </select>
      <select
        class="join-item select select-bordered w-full"
        id="variant"
        name="variant"
      >
        <option
          value="chess"
        >
          Chess
        </option>
        <option
          value="chess960"
        >
          Chess960
        </option>
      </select>
    </div>
  </header>
</div>
`;
